File: Shipfinity.Domain/Models/ProductOrder.cs
using System.ComponentModel.DataAnnotations;

namespace Shipfinity.Domain.Models
{
    public class ProductOrder : BaseEntity
    {
        public Order Order { get; set; }
        public int OrderId { get; set; }
        public Product Product { get; set; }
        public int ProductId { get; set; }
        [Range(0, 9999)]
        public int Quantity { get; set; }
    }
}

File: shipfinity.frontend/src/app/shared/models/product-order.ts
import Product from "./product";

export default class ProductOrder {
    product: Product;
    quantity: number;

    constructor(product?: Product, quantity?: number){
        this.product = !product? new Product() : product;
        this.quantity = !quantity? 1 : quantity;
        if(quantity === undefined || quantity <= 0){
            this.quantity = 1;
        }
    }

    setQuantity(q:number){
        if(q > 0){
            this.quantity = q;
        }
    }
}